[3.5] [Index Rollups] Add documentation for cardinality metric and multi-tier rollup in Index Rollups (#11828)
opensearch-project/documentation-website: _im-plugin/index-rollups/index.md

### Step 2: Define aggregations and metrics

> Select the attributes with the aggregations (terms and histograms) and metrics (avg, sum, max, min, value count, and cardinality (OpenSearch 3.5+)) that you want to roll up. Make sure you don’t add a lot of highly granular attributes, because you won’t save much space.

Author: Sure will remove this

    "enabled": true,
    "schedule": {
      "interval": {
        "period": 1,
        "unit": "Hours"
      }
    },
    "source_index": "web_logs",
    "target_index": "web_logs_hourly",
    "page_size": 1000,
    "dimensions": [
      {
        "date_histogram": {
          "source_field": "timestamp",
          "fixed_interval": "1h"
        }
      }
    ],
    "metrics": [
      {
        "source_field": "visitor_id",
        "metrics": [
          {
            "cardinality": {
              "precision_threshold": 10000
            }
          }
        ]
      },
      {
        "source_field": "page_views",
        "metrics": [
          {
            "sum": {}
          }
        ]
      }
    ]
  }
}
```

Query the rollup index:

```json
GET web_logs_hourly/_search
{
  "size": 0,
  "aggs": {
    "hourly_stats": {
      "date_histogram": {
        "field": "timestamp",
        "fixed_interval": "1h"
      },
      "aggs": {
        "unique_visitors": {
          "cardinality": {
            "field": "visitor_id"
          }
        },
        "total_page_views": {
          "sum": {
            "field": "page_views"
          }
        }
      }
    }
  }
}
```

> ## Multi-tier Rollups

Reviewer: This doesn't look like a lot of information yet. If there are additions to this, we can break this section out into a new page and link from this page. But if for now this is the only information, we can keep it here until additional functionality is implemented.

**Introduced 3.5**
{: .label .label-purple }

Multi-tier rollups allow you to create rollup-of-rollup jobs, enabling progressive data granularity reduction over time. This is useful for long-term data retention strategies where you want to keep fine-grained data for recent periods and increasingly coarse-grained data for older periods.

### Concept

With multi-tier rollups, you can create a hierarchy of rollup jobs:

```
Raw Data (5-second intervals)
    ↓ Tier 1 Rollup
Hourly Rollup (1-hour intervals)
    ↓ Tier 2 Rollup
Daily Rollup (1-day intervals)
    ↓ Tier 3 Rollup
Weekly Rollup (1-week intervals)
```

Each tier uses the previous tier's rollup index as its source, progressively reducing storage requirements while maintaining queryability.

### Requirements

For multi-tier rollups to work correctly:

1. **Matching dimensions**: All tiers must use the same dimension fields (field names and types must match)

> 2. **Compatible time intervals**: Each tier's interval must be a multiple of the previous tier's interval (e.g., 5m → 1h → 1d)

Reviewer 2: ```suggestion
2. **Compatible time intervals**: Each tier's interval must be a multiple of the previous tier's interval (for example, 5m → 1h → 1d).
```

3. **Consistent cardinality precision**: If using cardinality metrics, all tiers must use the same `precision_threshold` value
4. **Source index type**: The source index for tier 2+ must be a rollup index created by a previous tier

> ### Example: Two-tier Rollup Strategy

Reviewer 2: ```suggestion
### Example: Two-tier rollup strategy
```

With multi-tier rollups, you can create a hierarchy of rollup jobs:

```
Raw Data (5-second intervals)
    ↓ Tier 1 Rollup
Hourly Rollup (1-hour intervals)
    ↓ Tier 2 Rollup
Daily Rollup (1-day intervals)
    ↓ Tier 3 Rollup
Weekly Rollup (1-week intervals)
```

Each tier uses the previous tier's rollup index as its source, progressively reducing storage requirements while maintaining queryability.

### Prerequisites

For multi-tier rollups to work correctly, you must fulfill the following prerequisites:

1. **Matching dimensions**: All tiers must use the same dimension fields (field names and types must match).
2. **Compatible time intervals**: Each tier's interval must be a multiple of the previous tier's interval (for example, `5m` → `1h` → `1d`).
3. **Consistent cardinality precision**: If using [cardinality metrics](#cardinality-metric), all tiers must use the same `precision_threshold` value.
4. **Source index type**: The source index for tier 2+ must be a rollup index created by a previous tier.

> 5. **Source index data**: Ensure that the source tier has completed at least one rollup execution before creating the next tier. Creating a rollup job from an empty rollup index will succeed but may behave unexpectedly.

Reviewer 2: ```suggestion
5. **Source index data**: Ensure that the source tier has completed at least one rollup execution before creating the next tier. Creating a rollup job from an empty rollup index will succeed, but it may behave unexpectedly.
```
